src/olympia/devhub/views.py

```diff
@@ -1978,3 +1979,33 @@ def onboarding_subscription_cancel(request, addon_id, addon):
     return redirect(
         reverse("devhub.addons.onboarding_subscription", args=[addon.id])
     )
+
+
+@dev_required
+@post_required
+def subscription_customer_portal(request, addon_id, addon):
+    sub = get_promoted_subscription_or_404(addon=addon)
+
+    try:
+        session = retrieve_stripe_checkout_session(sub)
+    except Exception:
+        log.exception(
+            "error while trying to retrieve a stripe checkout session for "
+            "PromotedSubscription %s (customer_portal).",
+            sub.pk
+        )
+        raise http.Http404()
+
+    try:
+        portal = create_stripe_customer_portal(
```

Comment:
Just wondering, why not have `create_stripe_customer_portal` do the `retrieve_stripe_checkout_session`, rather than doing it above?

Reply:
This helps me return the right response/log the error differently.